### ISSUE TITLE
On-Call Jinja route support

### ISSUE THREAD

**Issue by nijave** [grafana/terraform-provider-grafana#849]
Grafana On-Call recently added support for using Jinja templates in routes https://github.com/grafana/oncall/pull/1319

This is useful since you can more exactly match labels coming from AlertManager derived alerts

**Comment by areh-dev** [comment]
Unfortunately, support for route type still doesn't exist, but I found some workaround:
**we can use CURL to change just created route by using (not recommended) provisioner local-exec**

```
resource "grafana_oncall_route" "route_to_sre" {
  integration_id      = grafana_oncall_integration.alertmanager_testing.id
  escalation_chain_id = grafana_oncall_escalation_chain.SRE.id
  position            = 0
  routing_regex       = "{{- payload.get('labels', {}).get('team', '') == 'SRE' }}"

  # routing_type = "jinja2" # wait for support by grafana terraform provider

  # (!) temporary curl crutch))
  provisioner "local-exec" {
    command = join(" ", [
      "curl ${local.oncall_api_url}/api/v1/routes/${self.id}/",
      "--request PUT",
      "--header 'Authorization: ${local.oncall_api_token}'",
      "--header 'Content-Type: application/json'",
      "--data '{ \"routing_type\": \"jinja2\" }'",
    ])
    quiet = true # to hide the API token as part of command to stdout
  }
}

locals {
  oncall_api_url     = "https://my-oncall.url"
  oncall_api_token   = "my-oncall-api-token"
}
```

**Comment by julienduchesne** [comment]
To be released in v1.42.0
